Friend/firmware/firmware_v1.0/src/transport.c

```diff
@@ -66,6 +70,34 @@ static struct bt_gatt_attr dfu_service_attr[] = {
 
 static struct bt_gatt_service dfu_service = BT_GATT_SERVICE(dfu_service_attr);
 
+
+
+static void button_ccc_config_changed_handler(const struct bt_gatt_attr *attr, uint16_t value);
+static ssize_t button_data_read_characteristic(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset);
+//button
+//
+//
+static struct gpio_callback button_cb_data;
+
+struct gpio_dt_spec d4_pin = {.port = DEVICE_DT_GET(DT_NODELABEL(gpio0)), .pin=4, .dt_flags = GPIO_OUTPUT_ACTIVE}; //3.3
+struct gpio_dt_spec d5_pin_input = {.port = DEVICE_DT_GET(DT_NODELABEL(gpio0)), .pin=5, .dt_flags = GPIO_INT_EDGE_RISING};
+
+static struct bt_uuid_128 button_uuid = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x00000003,0x0000,0x1000,0x7450,0xBE2E44B06B00));
+static struct bt_uuid_128 button_uuid_x = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x00000004,0x0000,0x1000,0x7450,0xBE2E44B06B00));
+
+static struct bt_gatt_attr button_service_attr[] = {
+    BT_GATT_PRIMARY_SERVICE(&button_uuid),
+    BT_GATT_CHARACTERISTIC(&button_uuid_x.uuid, BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY, BT_GATT_PERM_READ, button_data_read_characteristic, NULL, NULL),
+    BT_GATT_CCC(button_ccc_config_changed_handler, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+};
+
+static struct bt_gatt_service button_service = BT_GATT_SERVICE(button_service_attr);
```

Comment:
The button states don't appear to transfer when I add the characteristic to the audio service. I think space is an issue since the audio notify takes a lot of space? I think it's best if we leave this in a separate service for now.